list2/ex21/index.js

```diff
@@ -0,0 +1,42 @@
+import promptSync from 'prompt-sync';
+const prompt = promptSync();
+
+const WOMEN_WEIGHT_IDEAL_BASE = 62.1;
+const WOMEN_WEIGHT_FORMULA = 44.7;
+const MEN_WEIGHT_IDEAL_BASE = 72.7;
+const MEN_WEIGHT_FORMULA = 58;
+
+
+export function idealWeightFunction() {
+    const gender = {
+        male: 'M',
+        female: 'F'
+    }
+
+    console.log('What is the person gender? (Use M or F).');
+    const genderPrompt = prompt('>').toUpperCase();
+
+    if(!Object.values(gender).includes(genderPrompt)) {
+        console.log('Insert a valid gender.');
+        return;
+    }
+
+    console.log('What is the person height in meters ? (Use just float numbers).');
+    const heightPrompt = parseFloat(prompt('>'));
+
+    function idealWeightCalculation(height) {
+        let idealWeight;
+
+        if (genderPrompt === gender.male) {
+            idealWeight = (MEN_WEIGHT_IDEAL_BASE * height) - MEN_WEIGHT_FORMULA;
+        } else if (genderPrompt === gender.female) {
+            idealWeight = (WOMEN_WEIGHT_IDEAL_BASE * height) - WOMEN_WEIGHT_FORMULA;
+        }
+
+        return idealWeight;
+    }
```

Comment:
Not sure if you have learned anything about classes and design patterns already but we could get fancy here and apply the [Solid](https://en.wikipedia.org/wiki/SOLID) [Open/Closed Principle](https://en.wikipedia.org/wiki/SOLID#:~:text=%5B3%5D-,Open%E2%80%93closed%20principle,-%5Bedit%5D) by using the [Factory](https://refactoring.guru/design-patterns/factory-method) and [Strategy](https://refactoring.guru/design-patterns/strategy) patterns.  

```js
class AbstractIdealWeightCalculator {
  calculate(height) {
    throw new Error("This method should be overridden by concrete strategies")
  }
}

class ManIdealWeightCalculator extends AbstractIdealWeightCalculator {
  calculate(height) {
    (72.7 * height) - 58;
  } 
}

class WomanIdealWeightCalculator extends AbstractIdealWeightCalculator {
  calculate(height) {
    (62.1 * height) - 44.7;
  } 
}

class IdeadWeightCalculatorFactory {
  static createCalculator(calculatorType) {
    switch(calculatorType) {
      case "man":
        return new ManIdealWeightCalculator()
      case "woman":
        return new WomanIdealWeightCalculator()
      default: 
        throw new Error("Invalid calculator type")
    }
  }
}

const idealWeightCalculator = IdeadWeightCalculatorFactory.createCalculator(genderPrompt)
const idealWeight = idealWeightCalculator.calculate(height)
```

This would allow you to easily create new `IdealWeightCalculators` by simply extending your code and not having to change it much. 

NOTE: I'm not saying you should use it but just wanted to share so that you can take a look at the links and maybe learn something new :)

Reply:
BTW, let me know if you want to chat about it or have questions. I'm happy to assist!